### DISCUSSION TITLE
Question: How to override OnFileActivated?

### DISCUSSION THREAD

**Discussion by akpapa** [microsoft/WindowsAppSDK#785] [Q&A] [answered]
- Net 5 desktop app
- WinUI3 
- Project Reunion 0.5.5
- Windows 10 build 21370
- VS 2019 preview

I am trying to override OnFileActivated in App.xaml.cs to let my app open certain file type when double clicked.
But it seems there is no OnFileActivated base which allows me to override?
(OnFileActivated just does not show up after typed override keyword in VS2019)

this page says that I can do it with project reunion 0.5, anybody knows how?
https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.application.onfileactivated?view=winui-3.0-preview

Thanks!

**Comment by andrewleader** [comment, marked as answer]
Hi, thanks for asking!

The OnFileActivated method overload is only supported by UWP project types (which is only supported in the preview versions of Project Reunion). The docs do mention that for desktop apps, this method isn't called.

For desktop MSIX apps (what you created), today you can use `AppInstance.GetActivatedEventArgs()`: https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/get-activation-info-for-packaged-apps#code-example

We're additionally working on adding new AppLifecycle APIs for handling file activation (among other activations), you can see the spec here: https://github.com/microsoft/ProjectReunion/issues/111

AppLifecycle is planned to ship in 1.0, but we'll have some previews out in the 0.8 timeframe.

**Reply by akpapa** [reply]
Thank you @andrewleader for the information, it's very helpful!
Now `AppInstance.GetActivatedEventArgs()` works great!

I usually use Windows Forms and this is my first time using an xaml based platform, I pretty like it. 👍 
Really looking forward to the future releases especially I'm going to try some cool UI features like Acrylic.

Thank you guys for your amazing work!
